cedar-dafny/validation/thm/soundness.dfy

```diff
@@ -1178,6 +1179,51 @@ module validation.thm.soundness {
       }
     }
 
+    lemma SoundInSetMemberFalse(e1: Expr, ei2s: seq<Expr>, i: nat, effs: Effects)
+      decreases BinaryApp(BinaryOp.In,e1,Expr.Set(ei2s)) , 0 , Expr.Set(ei2s) , 0
+      requires InstanceOfRequestType(r,reqty)
+      requires InstanceOfEntityTypeStore(s,ets)
+      requires InstanceOfActionStore(s,acts)
+      requires EffectsInvariant(effs)
+      requires WellTyped(BinaryApp(BinaryOp.In,e1,Expr.Set(ei2s)),effs)
+      requires getType(BinaryApp(BinaryOp.In,e1,Expr.Set(ei2s)),effs) == Type.Bool(False)
+      requires !Typechecker(ets,acts,reqty).isUnspecifiedVar(e1)
+      requires 0 <= i < |ei2s|
+      ensures IsFalse(r,s,BinaryApp(BinaryOp.In,e1,ei2s[i]))
```

Comment:
Looks like this is a proof optimization unrelated to overflow validation? Would be cleaner to move to a separate PR